src/libwalletqt/Wallet.cpp

```diff
@@ -662,6 +662,26 @@ QString Wallet::getTxKey(const QString &txid) const
   return QString::fromStdString(m_walletImpl->getTxKey(txid.toStdString()));
 }
 
+void Wallet::getTxKeyAsync(const QString &txid, const QJSValue &ref) const
+{
+    QFuture<QString> future = QtConcurrent::run(this, &Wallet::getTxKey, txid);
+    auto watcher = new QFutureWatcher<QString>();
+
+    connect(watcher, &QFutureWatcher<QString>::finished,
+            this, [this, watcher, txid, ref]() {
+          QFuture<QString> future = watcher->future();
+          watcher->deleteLater();
+
+          auto txKey = future.result();
+          if (ref.isCallable()){
+              QJSValue cb(ref);
```

Comment:
Copy is required as param is a const reference and `call()` method is not const in Qt5 (there is a comment it should be const in Qt6). AFAIK this is usual practice.